src/goto-symex/goto_state.h

```diff
@@ -63,6 +63,20 @@ class goto_statet
   /// Threads
   unsigned atomic_section_id = 0;
 
+private:
+  std::size_t recorded_complexity = 0;
+
+public:
+  void record_complexity()
```

Comment:
It's not clear what is going on here. Is it meant to avoid recomputing guard size everytime? in that case use a `lazyt`, otherwise this probably shouldn't be part of this class

Reply:
It's cleaner to get the states to record their own complexity, and this method is just an abstraction as recording complexity can be anything, that it's so simple right now is just this particular implementation. Just saves time in the future for no real downside.

Reply:
> It's cleaner to get the states to record their own complexity

I'm not sure about that, if objects record information about the data that is in them, then it gets confusing. Right now I have no idea about what this complexity represents and when it should be updated. Is there a relation between this complexity and the other fields which should be preserved? If so, the interface of the class should ensure it is preserved.

Reply:
Shrunk it to what's really required right now, while still keeping in line with how it's meant to be used. If multiple recording/state is needed later it can be trivially put in anyway.